addons/mail/static/src/new/composer/composer.js

```diff
@@ -49,11 +49,11 @@ export class Composer extends Component {
     static template = "mail.composer";
 
     setup() {
+        this.messaging = useMessaging();
         this.attachmentUploader = useAttachmentUploader(
-            this.props.composer.thread,
+            this.messageToReplyTo?.originThread ?? this.props.composer.thread,
```

Comment:
Note that this is dangerous, because this could have made other composers not related to "replying" being affected to that.
The reply is canceled when clicking away from replying. This is not an issue thanks to this spec... But it would be wiser to properly contextualize `messageToReplyTo` in a follow-up PR, as chances to break code with small changes to reply-to feature is high